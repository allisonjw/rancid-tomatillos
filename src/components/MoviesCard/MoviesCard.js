import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import MovieRatings from '../../containers/MovieRatings/MovieRatings'
import '../../containers/App/App.scss';
import PropTypes from 'prop-types';

export default class MoviesCard extends Component {
  constructor() {
    super()
    this.state = {
      showPageOpen: false
    }
  }

  redirect = (e) => {
    e.preventDefault()
    this.setState({ showPageOpen: true })
  }

  render() {
    return (
      (this.state.showPageOpen)
      ? <Redirect to={`/movies/${this.props.id}`}/>
      : <section className="moviesContainer-movieCard-cardContainer">
          <h2 className="movieCard-cardContainer-title">{this.props.title}</h2>
          <img className="movieCard-cardContainer-poster" src={this.props.poster_path}alt="Official movie poster for film"></img>
          <h4>Average Movie Rating: {this.props.average_rating}</h4>
          <button className="movieCard-btn">Add Rating</button>
          <button className="movieCard-btn" onClick={(e) => this.redirect(e)}>See More</button>
          <MovieRatings />
        </section>
    )
  }
}



MoviesCard.propTypes = {
  average_rating: PropTypes.number,
  id: PropTypes.number,
  backdrop_path: PropTypes.string,
  overview: PropTypes.string,
  poster_path: PropTypes.string,
  release_date: PropTypes.string,
  title: PropTypes.string
}
