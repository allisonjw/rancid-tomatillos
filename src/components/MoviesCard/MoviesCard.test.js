/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import { MoviesCard } from './MoviesCard';

describe('MoviesCard', () => {
  const wrapper = shallow(<MoviesCard
    id = '1'
    title = "Jumanji: The Next Level"
    poster_path = "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg"
    backdrop_path = "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg"
    release_date = "2019-12-04"
    overview = "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game."
    average_rating = '4'
    user={{id: 7, name: 'Diana', email: "diana@turing.io"}}
       />);


  it('should match a snapshot of data being passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot of data being passed in if the user is logged out', () => {
    const wrapper = shallow(<MoviesCard
      id = '1'
      title = "Jumanji: The Next Level"
      poster_path = "https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg"
      backdrop_path = "https://image.tmdb.org/t/p/original//zTxHf9iIOCqRbxvl8W5QYKrsMLq.jpg"
      release_date = "2019-12-04"
      overview = "In Jumanji: The Next Level, the gang is back but the game has changed. As they return to rescue one of their own, the players will have to brave parts unknown from arid deserts to snowy mountains, to escape the world's most dangerous game."
      average_rating = '4'
      user={{}}
         />);
    expect(wrapper).toMatchSnapshot();
  });
});
