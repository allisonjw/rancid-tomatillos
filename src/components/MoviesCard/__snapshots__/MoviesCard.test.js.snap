// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesCard should match a snapshot of data being passed in 1`] = `
<section
  className="moviesContainer-movieCard-cardContainer"
>
  <h2
    className="movieCard-cardContainer-title"
  >
    Jumanji: The Next Level
  </h2>
  <img
    alt="Official movie poster for film"
    className="movieCard-cardContainer-poster"
    src="https://image.tmdb.org/t/p/original//l4iknLOenijaB85Zyb5SxH1gGz8.jpg"
  />
  <h4>
    Average Movie Rating: 
    4
  </h4>
  <button
    className="movieCard-btn"
  >
    Add Rating
  </button>
  <button
    className="movieCard-btn"
    onClick={[Function]}
  >
    See More
  </button>
  <MovieRatings />
</section>
`;
